test/e2e/network/loadbalancer.go

```diff
@@ -1236,6 +1244,143 @@ var _ = common.SIGDescribe("LoadBalancers", func() {
 			framework.Failf("Failed to connect to backend 2")
 		}
 	})
+
+	ginkgo.It("should not have connectivity disruption during rolling update [Slow]", func() {
+		ns := f.Namespace.Name
+		name := "test-lb-rolling-update"
+		labels := map[string]string{"name": name}
+		dsRetryPeriod := 1 * time.Second
+		dsRetryTimeout := 5 * time.Minute
+		gracePeriod := int64(10)
+		ds := e2edaemonset.NewDaemonSet(name, e2eapps.AgnhostImage, labels, nil, nil,
+			[]v1.ContainerPort{
+				{ContainerPort: 80},
+			},
+			"netexec", "--http-port=80", fmt.Sprintf("--delay-shutdown=%d", gracePeriod),
+		)
+		ds.Spec.UpdateStrategy = appsv1.DaemonSetUpdateStrategy{
+			Type: appsv1.RollingUpdateDaemonSetStrategyType,
+		}
+		ds.Spec.Template.Labels = labels
+		ds.Spec.Template.Spec.TerminationGracePeriodSeconds = utilpointer.Int64(gracePeriod)
+
+		nodeNames := e2edaemonset.SchedulableNodes(cs, ds)
+		e2eskipper.SkipUnlessAtLeast(len(nodeNames), 2, "load-balancer rolling update test requires at least 2 schedulable nodes for the DaemonSet")
+
+		ginkgo.By(fmt.Sprintf("Creating DaemonSet %q", name))
+		ds, err = cs.AppsV1().DaemonSets(ns).Create(context.TODO(), ds, metav1.CreateOptions{})
+		framework.ExpectNoError(err)
+
+		ginkgo.By("Checking that daemon pods launch on every schedulable node of the cluster")
+		err = wait.PollImmediate(dsRetryPeriod, dsRetryTimeout, e2edaemonset.CheckDaemonPodOnNodes(f, ds, nodeNames))
+		framework.ExpectNoError(err, "error waiting for daemon pods to start")
+		err = e2edaemonset.CheckDaemonStatus(f, name)
+		framework.ExpectNoError(err)
+
+		ginkgo.By(fmt.Sprintf("Creating a service %s with type=LoadBalancer in namespace %s", name, ns))
+		jig := e2eservice.NewTestJig(cs, ns, name)
+		jig.Labels = labels
+		service, err := jig.CreateLoadBalancerService(e2eservice.GetServiceLoadBalancerCreationTimeout(cs), func(svc *v1.Service) {})
```

Comment:
I think we should have two versions of this test, one with `externalTrafficPolicy=Cluster` (this test) and another with `externalTrafficPolicy=Local`. Thoughts @ionutbalutoiu @aojea?

Reply:
I was thinking about this. We should definitely do this. I will add it.

Reply:
❤️

Reply:
Updated the PR to include the two tests.